### DISCUSSION TITLE
Modal component to have a configurable behaviour on pressing Escape key

### DISCUSSION THREAD

**Discussion by ijpatricio** [filamentphp/filament#11161] [Ideas]
Hey team Filament!

On a project, I just needed a modal that wouldn't close by pressing Escape Key.

I ended up publishing the blade for the `modal` component, and implemented such thing. Check picture below.

I've read https://github.com/filamentphp/filament/discussions/3089 but I do think if configurable is good for developing with Filament's UI outside of panels.

Is this something you would like a PR for? I do think it is, but wanted to check in first 😄 

Thanks!

![image](https://github.com/filamentphp/filament/assets/26031459/5124dd1c-6ded-407e-8da7-4a5fcc3f930d)

**Comment by myregistration** [comment]
I was researching to see if there is a way to set a listener for escape key on persistent Notification and saw your post
